### DISCUSSION TITLE
Can not find "runInTransaction" param when I use extension.

### DISCUSSION THREAD

**Discussion by BarrySJL** [prisma/prisma#20708] [Q&A] [answered]
### Question

I want to use extension to instead of middleware. But I could not find  the "runInTransaction" param in [Prisma Client extensions query component type]. What's the solution for this?

<img width="593" alt="Screenshot 2023-08-16 at 17 32 00" src="https://github.com/prisma/prisma/assets/26754558/06579e57-0f26-4fc2-8290-37e076b3cfdd">

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @BarrySJL 👋 

The `runInTransaction` parameter is not directly available in the Prisma Client extensions query component type. However, you can use the `$transaction` method to handle transactions in your extensions. See this [section](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/query#wrap-a-query-into-a-batch-transaction) of the documentation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by BarrySJL** [reply]
Thanks for your answer. but it does not satisfy demand for the product. we just want this parameter  tell this operation is in a transaction or not. like middleware.

Does Prisma will support this in the future?

**Reply by ludralph** [reply, marked as answer]
Hi @BarrySJL, I just confirmed from the engineering team that we will provide support for this in the future in some form or another.

If this answers your question, it would be great if you could mark this Discussion as answered to indicate that it has been resolved.

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
